### PR TITLE
[Program:GCI] Added missing tests to the accept request API.

### PR DESCRIPTION
**Pull request by rudrakshjhalanii07**
### Description
Included the tests which were mentioned in the task successfully.

Fixes # [ISSUE]
#306 

### Type of Change:
- Code

**Code/Quality Assurance Only**
- New feature (non-breaking change which adds functionality pre-approved by mentors)

### How Has This Been Tested?
All the tests have been tested by me manually.

### Checklist:
**Delete irrelevant options.**

- [x] My PR follows the style guidelines of this project
- [x] I have performed a self-review of my own code or materials
- [x] I have commented my code or provided relevant documentation, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] Any dependent changes have been merged
- [ ] Update Postman API at /docs folder
- [ ] Update Swagger documentation and the exported file at /docs folder
- [ ] Update requirements.txt

**Code/Quality Assurance Only**
- [x] My changes generate no new warnings 
- [ ] My PR currently breaks something (fix or feature that would cause existing functionality to not work as expected)
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [ ] Any dependent changes have been published in downstream modules

### CODE REVIEW

**Review by Tlazypanda — CHANGES_REQUESTED** [review]
Test cases 2,3 and 4 missing from [table](https://github.com/systers/mentorship-backend/wiki/Quality-Assurance#accept-relation).Can you add those tests?

**Comment by rudrakshjhalanii07** [comment]
Oh! I forgot to add those.
I shall just add them up.

**Review by Tlazypanda — CHANGES_REQUESTED** [review]
@Hardik0403 Umm...still can't find those tests :sweat_smile:

**Comment by rudrakshjhalanii07** [comment]
Sorry for the confusion, I have updated the PR though!

File: app/api/dao/mentorship_relation.py [hunk 1]
```diff
@@ -183,6 +183,10 @@ def accept_request(user_id, request_id):
             if request.state == MentorshipRelationState.ACCEPTED:
                 return messages.USER_IS_INVOLVED_IN_A_MENTORSHIP_RELATION, 400
 
+        # verify if user is involved in other mentorship relation
```
Comment (Tlazypanda) [inline app/api/dao/mentorship_relation.py]: This is not required `USER_IS_INVOLVED_IN_A_MENTORSHIP_RELATION` is already there

File: tests/mentorship_relation/test_dao_accept_request.py [hunk 2]
```diff
@@ -53,16 +53,31 @@ def test_dao_requester_tries_to_accept_mentorship_request(self):
 
         result = DAO.accept_request(self.first_user.id, self.mentorship_relation.id)
 
-        self.assertEqual((messages.CANT_ACCEPT_MENTOR_REQ_SENT_BY_USER, 400), result)
+        # If a user tries to accepts a request where he is not involved, then
```
Comment (Tlazypanda) [inline tests/mentorship_relation/test_dao_accept_request.py]: Changes in this file not required for now

**Review by Tlazypanda — CHANGES_REQUESTED** [review]
@Hardik0403 Awesome work! :100: Just a few more changes and then ready to be approved

File: tests/mentorship_relation/test_dao_accept_request.py (hunk 2 above)
Comment (rudrakshjhalanii07) [reply tests/mentorship_relation/test_dao_accept_request.py]: Wow, didn't expected that the tests would run without these changes!😅
I have updated the PR with lesser changes.😄

File: app/api/dao/mentorship_relation.py (hunk 1 above)
Comment (rudrakshjhalanii07) [reply app/api/dao/mentorship_relation.py]: Updated!

**Review by Tlazypanda — APPROVED** [review]
Approved :) But can you squash the commits into one?

**Comment by isabelcosta** [comment]
@Hardik0403  thank you for your contribution here 🎉
I will close this PR because we decided to merge another GCI PR.

**Pull request closed without merging**